scripts/fetchODCDataFilesUtils.js

```diff
@@ -0,0 +1,172 @@
+	 * @return {Boolean} true if the file name is already present in the data folder, false if not
+	 */
+	fileNotExists: (dataFileName, dataFileDir) => {
+		let allFiles = fs.readdirSync(dataFileDir);
+		return !allFiles.includes(dataFileName);
+	},
+
+	/**
+	 * Download a file from the given download URL and write into a target local file.
+	 * @param {String} downloadURL The download URL
+	 * @param {String} targetFileLocation The target file location including the path and file name
+	 */
+	downloadDataFile: async (downloadURL, targetFileLocation) => {
+		try {
+			let res = await axios.get(downloadURL);
+			fs.writeFileSync(targetFileLocation, res.data, "utf8");
+		} catch (error) {
+			// do nothing
+		}
+	},
+
+	exitWithError: (err, clonedLocalDir) => {
+		rimraf.sync(clonedLocalDir);
+		console.log(err);
+		process.exit(1);
+	},
+
+	// Clone the COVID data repository and push the new data file to a remote branch
+	prepareLocalRepo: async (covidDataRepoUrl, clonedLocalDir, wecountprojectRepoUrl) => {
+		await git.clone(covidDataRepoUrl, clonedLocalDir)
+			.cwd(clonedLocalDir)
+			.addRemote("wecountproject", wecountprojectRepoUrl)
+			.catch((err) => module.exports.exitWithError(err, clonedLocalDir));
+	},
+
+	// Clone the COVID data repository and push the new data file to a remote branch
+	createRemoteBranch: async (branchName, publishedDate, clonedLocalDir) => {
+		await git.checkoutLocalBranch(branchName)
+			.add("./*")
+			.commit("feat: commit a new ODC data file published on " + publishedDate)
+			.push("wecountproject", branchName)
+			.catch((err) => module.exports.exitWithError("Error at pushing to a remote branch named " + branchName + ". Check either the authentication or whether the remote branch " + branchName + " already exists.\n" + err, clonedLocalDir));
+	},
+
+	issuePullRequest: async (githubAPI, covidDataRepoUrl, accessToken, branchName, publishedDate) => {
+		// Parse out the owner and repo name that the pull request will be issued for
+		const pattern = /https:\/\/github.com\/(.*)\/(.*).git/;
+		const matches = pattern.exec(covidDataRepoUrl);
+		const headers = {
+			"Authorization": "bearer " + accessToken
+		};
+
+		const repoInfoResponse = await axios.post(githubAPI, {
+			query: `query {
+			repository(owner: "${matches[1]}", name: "${matches[2]}") {
+				url
+				id
+			}
+			}`
+		}, {
+			headers:headers
+		});
+
+		const repoId = repoInfoResponse.data.data.repository.id;
+
+		const createPRResponse = await axios.post(githubAPI, {
+			query: `mutation {
+			createPullRequest (
+				input: {
+					baseRefName:"main",
+					headRefName: "wecountproject:${branchName}",
+					repositoryId: "${repoId}",
+					title: "Add a new ODC data file published on ${publishedDate}"
+				}) {
+				pullRequest {
+				  url
+				}
+			  }
+			}`
+		}, {
+			headers:headers
+		});
+
+		if (createPRResponse.data.errors) {
+			console.log("Error at creating the new pull request: ", createPRResponse.data.errors);
+			process.exit(1);
```

Comment:
Again, rather than side-effect, better to raise an error payload and let the driver determine what to do with it